crates/task-impls/src/request.rs

```diff
@@ -191,11 +200,39 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> NetworkRequestState<TYPES, I
             return;
         };
         debug!("Requesting data: {:?}", request);
-        let handle = async_spawn(requester.run(request, signature, self.public_key.clone()));
+        let handle = async_spawn(requester.run(request, signature));
 
         self.spawned_tasks.entry(view).or_default().push(handle);
     }
 
+    /// Spawns a task to send a request for the proposal and send the response on a channel
+    fn run_proposal(
+        &mut self,
+        request: &RequestKind<TYPES>,
+        response_chan: Sender<Option<Proposal<TYPES, QuorumProposal<TYPES>>>>,
+        view: TYPES::Time,
+    ) {
+        let leader = self.da_membership.leader(view);
+        let requester = ProposalRequester::<TYPES, I> {
+            network: Arc::clone(&self.network),
+            sender: response_chan,
+            leader,
+        };
+        let Ok(data) = bincode::serialize(&request) else {
+            tracing::error!("Failed to serialize request!");
+            return;
+        };
+        let Ok(signature) = TYPES::SignatureKey::sign(&self.private_key, &Sha256::digest(data))
+        else {
+            error!("Failed to sign Data Request");
+            return;
+        };
```

Comment:
Nit: Looks like `do_proposal` will do another serialization on these parameters. Can we put these 4 serializations in one function?

Reply:
I combined the sign and serialize part into a function used in both code paths.  The serialization in `do_proposal` is different as it's serializing the entire `Message` including the signature to be sent over the wire so it can't be combined with this serialization